packages/ibm-products/src/components/CoachmarkStack/CoachmarkStackHome.tsx

```diff
@@ -96,183 +95,170 @@ const componentName = 'CoachmarkStackHome';
+    if (node && navItemRefs.current[index] !== node) {
+      const isOverflowing = node.scrollWidth > node.clientWidth;
+      navItemRefs.current[index] = node;
+      setOverflowStates((prev) => {
+        const newState = [...prev];
+        newState[index] = isOverflowing;
+        return newState;
+      });
+    }
+  };
+
+  function renderNavLink(
+    index,
+    label,
+    ref: React.RefObject<ButtonProps<React.ElementType>> | null = null
+  ) {
+    const isOverflowing = overflowStates[index] ?? false;
 
-      return (
-        <li
-          key={index}
-          ref={(node) => {
-            itemRefHandler(index, node);
+    return (
+      <li
+        key={index}
+        ref={(node) => {
+          itemRefHandler(index, node);
+        }}
+      >
+        <Button
+          kind="ghost"
+          size="sm"
+          onClick={() => {
+            setLinkFocusIndex(index);
+            onClickNavItem(index + 1);
           }}
+          ref={ref}
         >
-          <Button
-            kind="ghost"
-            size="sm"
-            onClick={() => {
-              setLinkFocusIndex(index);
-              onClickNavItem(index + 1);
-            }}
-            ref={ref}
-          >
-            {isOverflowing ? (
-              <Tooltip
-                highContrast={false}
-                label={label}
-                align={tooltipAlign}
-                className={`${blockClass}__navLinkLabels-tooltip`}
-              >
-                <span className={`${blockClass}__navLinkLabels-text`}>
-                  {label}
-                </span>
-              </Tooltip>
-            ) : (
-              label
-            )}
-          </Button>
-        </li>
-      );
-    }
+          {isOverflowing ? (
+            <Tooltip
+              highContrast={false}
+              label={label}
+              align={tooltipAlign}
+              className={`${blockClass}__navLinkLabels-tooltip`}
+            >
+              <span className={`${blockClass}__navLinkLabels-text`}>
+                {label}
+              </span>
+            </Tooltip>
+          ) : (
+            label
+          )}
+        </Button>
+      </li>
+    );
+  }
 
-    return portalNode?.current
-      ? createPortal(
-          <div
-            {
-              // Pass through any other property values as HTML attributes.
-              ...rest
-            }
-            className={cx(blockClass, className)}
-            ref={ref}
-            {...getDevtoolsProps(componentName)}
-          >
-            <CoachmarkHeader
-              onClose={() => {
-                setLinkFocusIndex(0);
-                onClose();
-              }}
-            />
-            <div className={`${overlayClass}__body`}>
-              <div className={`${overlayClass}-element`}>
-                {!renderMedia && (
-                  <Idea size={20} className={`${blockClass}__icon-idea`} />
-                )}
+  return createPortal(
```

Comment:
The [Next.js portal example](https://github.com/vercel/next.js/blob/canary/examples/with-portals/components/ClientOnlyPortal.js) closer aligns to what we currently have where a portal is conditionally rendered if the `portalNode` ref (or some value tracking if the component has mounted) has been set.

Reply:
ok let me take another look at this 👍